### PR TITLE
Monoidal streams

### PR DESCRIPTION
**Pull request by toumix**
- [x] implement `id, then, tensor, swap, copy`
- [x] add documentation with pictures for unrolling
- [x] add documentation with pictures for the axioms
  - [x] associativity of tensor and composition
  - [x] interchange law
  - [x] feedback axioms
- [x] make the Fibonacci test pass
- [x] add the random walk as a test, make it pass

### CODE REVIEW

File: test/semantics/stream.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    s.check_later()
+    s.unroll().check_later()
+    assert s.feedback(T(x), T(y), T(m)).unroll(2).now(1, 2, 3) == (
+        True, False, True, '1 2 3')
+
+x, y, z, m, n = map(Ty.sequence, "xyzmn")
+x_, y_, z_, m_, n_ = [Ty.sequence(symmetric.Ty(name + "_")) for name in "xyzmn"]
+f, g = Stream.sequence('f', x, y, m), Stream.sequence('g', y, z, n)
+f_, g_ = Stream.sequence('f_', x_, y_, m_), Stream.sequence('g_', y_, z_, n_)
+LHS, RHS = (f >> g) @ (f_ @ g_), f @ f_ >> g @ g_
```
Comment (boldar99) [inline test/semantics/stream.py:39]: And these ones should be inside a test

File: discopy/feedback.py [hunk 2]
```diff
@@ -455,27 +544,26 @@ def __repr__(self):
 
 class FollowedBy(Box):
     """ The isomorphism between `x.head @ x.tail.delay()` and `x`. """
```
Comment (boldar99) [inline discopy/feedback.py:546]: We could add a simple example

Comment (toumix) [reply discopy/feedback.py]: Good idea, this one is not so intuitive!

File: test/semantics/stream.py (hunk 1 above)
Comment (toumix) [reply test/semantics/stream.py]: That was junk I removed it

File: test/semantics/stream.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    now = python.Function(lambda n: (bool(n % 2), str(n)), (x, ), (y, m))
+    later = python.Function(
+        lambda n, s: (bool(n % 2), s + " " + str(n)), (x, m), (y, m))
+    s = S(now, dom, cod, _later=lambda: S(later))
+    s.check_later()
+    s.unroll().check_later()
+    assert s.feedback(T(x), T(y), T(m)).unroll(2).now(1, 2, 3) == (
+        True, False, True, '1 2 3')
+
+x, y, z, m, n = map(Ty.sequence, "xyzmn")
```
[the remark this answers is not in the document]
Comment (toumix) [reply test/semantics/stream.py]: I removed this too

**Pull request merged**